src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs

```diff
@@ -789,5 +789,13 @@ internal enum GCConfigurationType
 
         [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "GCInterface_EnumerateConfigurationValues")]
         internal static unsafe partial void _EnumerateConfigurationValues(void* configurationDictionary, delegate* unmanaged<void*, void*, void*, GCConfigurationType, long, void> callback);
+
+        private static int _RefreshMemoryLimit()
+        {
+            return RefreshMemoryLimit();
```

Comment:
See like making the `AppConfig` check here would be more efficient than doing the work in native. The complex querying can all be done directly here instead of in the VM. Not terribly important for an experiment but it would be the preferred approach I think.

Reply:
That's what I am doing right now. By the time we are actually refreshing the memory, I suspended the runtime so calling managed code is impossible.